### ISSUE TITLE
[Bug]: Stable release moved exports into non-type bucket as types

### ISSUE THREAD

**Issue by Mitch-At-Work** [microsoft/fluentui#32968]
### Area

nx

### Environment

```shell
System:
    OS: Windows 11 10.0.22635
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H
    Memory: 17.86 GB / 31.69 GB
  Binaries:
    Node: 20.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.23.34 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    eslint: 8.57.0 => 8.57.0
    prettier: 2.8.8 => 2.8.8
    storybook: 7.6.20 => 7.6.20
    typescript: 5.0.4 => 5.0.4
```

### Current Behavior

When migrating carousel to stable, one classes types were moved to the non-type export (with the inline type modifier):
https://github.com/microsoft/fluentui/pull/32883/files/ca94657a665b1cb73a41d77dc4b852c0ab8073e2#r1783946352

### Expected Behavior

Types should be correctly exported in the aggregated type object.

### Reproduction

https://github.com/microsoft/fluentui

### Steps to reproduce

1.) Load master branch prior to react-carousel stable version: 5bea3fa901c841e41438fff4b7ae6b38bf30b982
2.) Release carousel to stable via script: `yarn nx g @fluentui/workspace-plugin:prepare-initial-release --project @fluentui/react-carousel-preview --phase=stable`
3.) CarouselSlider types in 'packages/react-components/react-components/src/index.ts' are in wrong export object

### Suggested severity

Medium - Has workaround

### Products/sites affected

Potentially exporting types incorrectly, not breaking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Hotell** [comment]
thanks for report ! 

after taking additional look the automation works as expected
> how it works in nutshell: it parses package barel AST, takes all export and export type nodes , updates react-components barrel AST and emits it, meaning -> exports are "copy pasted" from the sub-package source to suite source "as is".

The issue is within carousel barrel file which contains following:

<img width="602" alt="image" src="https://github.com/user-attachments/assets/e68f654d-f2ac-4c83-b014-37b291ea74c1">

This should be fixed once this lands https://github.com/microsoft/fluentui/issues/31350.

I'm gonna mark this "works as expected" as there is no bug per see.

ty!

**Comment by TristanWatanabe** [comment]
Closing this as @Hotell said this currently works as expected. A new issue has been created that supersedes this #31350
